data/json/effects_on_condition/npc_eocs/isherwood_barry_rescue_eocs.json

```diff
@@ -0,0 +1,328 @@
+        "then": [
+          {
+            "mapgen_update": "nest_Isherwood_firing_wall_add",
+            "om_terrain": "rural_road",
+            "om_special": "Barry's Mi-Go Scout Tower"
+          }
+        ]
+      },
+      { "run_eocs": [ "isherwood_mission_chris_spawn_eoc", "isherwood_mission_lisa_spawn_eoc" ] },
+      {
+        "u_location_variable": { "global_val": "invasion_map_proper_place" },
+        "terrain": "t_dirt_isherwood_player_spawnpoint",
+        "target_max_radius": 60,
+        "min_radius": 0,
+        "max_radius": 0
+      },
+      { "u_teleport": { "global_val": "invasion_map_proper_place" }, "force": true }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_wall_finished",
+    "effect": [ { "math": [ "isherwood_wall_status", "=", "1" ] } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_lisa_coming",
+    "effect": [ { "math": [ "isherwood_lisa_coming", "=", "1" ] } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_barry_return_home",
+    "global": true,
+    "eoc_type": "EVENT",
+    "required_event": "avatar_enters_omt",
+    "condition": {
+      "and": [
+        { "u_has_mission": "MISSION_ISHERWOOD_CHRIS_1" },
+        { "u_has_var": "barry_following", "type": "general", "context": "meeting", "value": "yes" },
+        { "u_near_om_location": "horse_farm_isherwood_13", "range": 17 }
+      ]
+    },
+    "effect": [
+      { "finish_mission": "MISSION_ISHERWOOD_CHRIS_1", "success": true },
+      {
+        "u_message": "Barry turns about and tells you, \"I know where I am now, I'll go on ahead.  They'll be so happy to see me, thank you for everything!  Come drop by the farm sometime, we'll have something for you!\"",
+        "type": "good",
+        "popup": true
+      },
+      {
+        "u_run_npc_eocs": [
+          {
+            "id": "EOC_BARRY_LEAVE",
+            "effect": [
+              {
+                "u_location_variable": { "global_val": "invasion_map" },
+                "target_params": { "om_terrain": "barry_mi-go_scout_tower_1", "search_range": 500, "z": 0 }
+              },
+              { "u_teleport": { "global_val": "invasion_map" }, "force": true },
+              {
+                "mapgen_update": "remove_barry_and_company",
+                "om_terrain": "barry_mi-go_scout_tower_1",
+                "om_special": "Barry's Mi-Go Scout Tower"
+              }
+            ]
+          }
+        ],
+        "unique_ids": [ "BARRY_MISSION", "CHRIS_MISSION", "LISA_MISSION" ]
+      }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "isherwood_mission_large_damage_retreat",
+    "eoc_type": "EVENT",
+    "required_event": "character_takes_damage",
+    "//": "I cant get this one to work right now, and I have no idea what's wrong.  It will need to be fixed.",
+    "condition": { "and": [ { "math": [ "currently_invading_map", "==", "1" ] } ] },
```

Comment:
You've got an 'and' here but only a single condition. That may be part of your issue, was there more logic you planned to include?

Reply:
I fixed it!